src/test/samVisualize/graphGeneration/graphGeneration.test.ts

```diff
@@ -0,0 +1,106 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as assert from 'assert'
+import { generateGraphFromYaml } from '../../../samVisualize/graphGeneration/cfnTemplateGraphGenerator'
+import { Node, Link } from '../../../samVisualize/graphGeneration/graph'
+import { join } from 'path'
+import { readdirSync, readFileSync } from 'fs'
+import { getProjectDir } from '../../testUtil'
+
+const TestInputDirectory = join(
+    getProjectDir(),
+    'testFixtures',
+    'workspaceFolder',
+    'samVisualize-test-data',
+    'testYamlTemplates'
+)
+const TestExpectedOutputDirectory = join(
+    getProjectDir(),
+    'testFixtures',
+    'workspaceFolder',
+    'samVisualize-test-data',
+    'expectedGraphObjects'
+)
+
+let inputFiles: Array<string>
+let expectedOutputFiles: Array<string>
+
+describe('samVisualize Graph Generation from YAML', async function () {
+    /**
+     * Returns the name of a file without an extension
+     * @param file File name from which the extension is removed
+     * @returns A string file name with the extension removed
+     */
+    function withoutExtension(file: string) {
+        return file.replace(/\.[^/.]+$/, '')
+    }
+
+    /**
+     * Sorts Node objects lexigraphically by name.
+     * No two Nodes in a graph will have the same name (no need to compare Node type)
+     */
+    function nodeAlphaCompare(n1: Node, n2: Node) {
+        return n1.name > n2.name ? 1 : -1
+    }
+
+    /**
+     * Sorts Link objects lexigraphically first by source, then by target, and finally by type.
+     */
+    function linkAlphaCompare(l1: Link, l2: Link) {
+        if (l1.source === l2.source) {
+            if (l1.target === l2.target) {
+                if (l1.type === l2.type) {
+                    return 0
+                }
+                // undefined gets placed after any value
+                if (l1.type === undefined) {
+                    return -1
+                }
+                if (l2.type === undefined) {
+                    return 1
+                }
+                return l1.type > l2.type ? 1 : -1
+            }
+            return l1.target > l2.target ? 1 : -1
+        }
+        return l1.source > l2.source ? 1 : -1
+    }
+
+    before(function () {
+        inputFiles = readdirSync(TestInputDirectory)
+        expectedOutputFiles = readdirSync(TestExpectedOutputDirectory)
+    })
+
+    it('same number of input files and expected output files', function () {
```

Comment:
I'd move this to an assert in the test below. It's not really testing any functionality; just add a message as a third param to the strictEqual statement.

Reply:
Makes sense, will do